src/lib.rs

```diff
@@ -375,7 +375,13 @@ impl CIPlatform {
 
 fn get_build_deps(manifest_location: &path::Path) -> io::Result<Vec<(String, String)>> {
     let mut lock_buf = String::new();
-    fs::File::open(manifest_location.join("Cargo.lock"))?.read_to_string(&mut lock_buf)?;
+    let path: PathBuf;
+    if cfg!(feature = "workspace") {
+        path = manifest_location.join("../Cargo.lock")
```

Comment:
This is potentially platform-incompatible. There is `ancestors()` and `parent()` to handle this is a platform-agnostic way.